setup.py

```diff
@@ -186,4 +186,5 @@ def _decode(x):
         "numba>=0.48.0",
     ],
     python_requires=">=3.6",
+    extras_require={"dev": ["pytest", "pre-commit", "black"],},
```

Comment:
We could add `networkx` and `python-igraph` as `dev` requirements here. The tests fail to run if those modules are not installed, but they are (at the moment) not needed otherwise.

Reply:
Yes I agree!